src/lib.rs

```diff
@@ -996,6 +996,49 @@ impl Build {
             }
         }
 
+        if self.cuda && cfg!(feature = "cudart") {
+            match which(&self.get_compiler().path) {
+                None => (),
+                Some(nvcc) => {
+                    // Try to figure out the -L search path. If it fails,
```

Comment:
Out of curiosity, do you know why `nvcc` doesn't do this logic automatically? Is there documentation we can point to somewhere recommending this `-L` path and such?

Reply:
It does do this automatically, but the problem is that `cargo` doesn't call `nvcc` to link the final binary. Instead is calls the system linker, and then you have to craft -L and -l yourself. Of course ideal would be if one could query `nvcc` and ask "what's your library path," but there is no such option:-(

Hmmm, when I think about it... **If** there is a way to dynamically override the target `linker`, and set it to `nvcc`, it might work too... In such case one wouldn't need this -L thing, but then cross-compilation might get tricky. [Just in case, reference to "cross-compilation" means that it does work, I can cross-compile CUDA program for aarch64.]

Reply:
The final linker is actually invoked by rustc rather than Cargo, and the target's default can be overridden with `-Clinker=/path/to/nvcc` (which can also be configured in cargo either via `RUSTFLAGS` or as a string with `CARGO_TARGET_..._LINKER=...`)

Is this a case where the Rust target should be defaulting to `nvcc` rather than `cc`?

(I should point out again I'm very naive about CUDA/nvcc/etc, I naively assume right now that there's one Rust target that everyone uses for GPU things (and that nvcc is for GPU things) and that you're compiling for that target)

Reply:
> Is this a case where the Rust target should be defaulting to `nvcc` rather than `cc`?

Yes. Or rather that's what I had in mind, it remains to be seen if it actually works all the way.

Reply:
> I naively assume right now that there's one Rust target that everyone uses for GPU things (and that nvcc is for GPU things) and that you're compiling for that target.

nvcc is available for x86_64, arm64 and ppc64le, And it can do cross-compilations. It's most common scenario when you run eclipse on x86_64 and target remote arm64.

Reply:
Setting LINKER to nvcc doesn't work, because it doesn't recognize all system's linker flags.

Reply:
Oh it might be the case that rustc needs more invasive knowledge about nvcc if it runs nvcc, there's a few flavors of linkers implemented in rustc and nvcc may need its own flavor (unless it looks exactly like gcc, which sounds like that's not the case)